client/trino-cli/src/test/java/io/trino/cli/TestClientRedirect.java

```diff
@@ -0,0 +1,191 @@
+import static io.trino.client.ClientStandardTypes.BIGINT;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_METHOD;
+
+@TestInstance(PER_METHOD)
+public class TestClientRedirect
+{
+    private static final JsonCodec<QueryResults> QUERY_RESULTS_CODEC = jsonCodec(QueryResults.class);
+    private static final String DUMMY_ACCESS_TOKEN = Base64.getEncoder().encodeToString("jwt-signing-key".getBytes(US_ASCII));
+    private MockWebServer server;
+
+    @BeforeEach
+    public void setup()
+            throws IOException
+    {
+        server = new MockWebServer();
+        server.start();
+    }
+
+    @AfterEach
+    public void teardown()
+            throws IOException
+    {
+        server.close();
+        server = null;
+    }
+
+    @Test
+    public void testDisableClientRedirectsDefault()
+            throws Exception
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("redirect/v1/statement"))
+                .addHeader(AUTHORIZATION, DUMMY_ACCESS_TOKEN));
+        server.enqueue(new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(createResults(server)));
+
+        QueryRunner queryRunner = createQueryRunner(createTrinoUri(server, Optional.empty()), createClientSession(server));
+
+        try (Query query = queryRunner.startQuery("SELECT 1")) {
+            query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
+        }
+
+        //verify the redirect request has the same Authorization header
+        RecordedRequest redirectedRequest = server.takeRequest();
+        assertThat(redirectedRequest.getHeader(AUTHORIZATION)).isEqualTo("Bearer " + DUMMY_ACCESS_TOKEN);
+    }
+
+    @Test
+    public void testClientRedirectEnabled()
+            throws Exception
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("redirect/v1/statement"))
+                .addHeader(AUTHORIZATION, DUMMY_ACCESS_TOKEN));
+        server.enqueue(new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(createResults(server)));
+
+        QueryRunner queryRunner = createQueryRunner(createTrinoUri(server, Optional.of(false)), createClientSession(server));
+
+        try (Query query = queryRunner.startQuery("SELECT 1")) {
+            query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
+        }
+
+        //verify the redirect request has the same Authorization header
+        RecordedRequest redirectedRequest = server.takeRequest();
+        assertThat(redirectedRequest.getHeader(AUTHORIZATION)).isEqualTo("Bearer " + DUMMY_ACCESS_TOKEN);
```

Comment:
The test in this function should be exactly the same as `testDisableClientRedirectsDefault()` except QueryRunner.
Extract function to prevent code duplication.